distribution/xfer/download.go

```diff
@@ -323,11 +323,7 @@ func (ldm *LayerDownloadManager) makeDownloadFunc(descriptor DownloadDescriptor,
 			if fs, ok := descriptor.(distribution.Describable); ok {
 				src = fs.Descriptor()
 			}
-			if ds, ok := d.layerStore.(layer.DescribableStore); ok {
-				d.layer, err = ds.RegisterWithDescriptor(inflatedLayerData, parentLayer, src)
-			} else {
-				d.layer, err = d.layerStore.Register(inflatedLayerData, parentLayer)
-			}
+			d.layer, err = d.layerStore.Register(inflatedLayerData, parentLayer, src)
```

Comment:
@stevvooe @aaronlehmann as part of what said in earlier comments - we should store the compressed layer https://github.com/docker/docker/pull/23014/files#diff-0c4769e1568c0f439863abd30df1ceccL313 along with the descriptor - do we want to create a `descriptorStore` or it's layerStore's business to store descriptor and compressed layer? (for later invalidation possibly)